flask_resources/serializers/csv.py

```diff
@@ -0,0 +1,229 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Flask-Resources is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""CSV serializer."""
+
+import csv
+
+from .base import BaseSerializer
+
+
+class Line(object):
+    """Object that implements an interface the csv writer accepts."""
+
+    def __init__(self):
+        """Initialize."""
+        self._line = None
+
+    def write(self, line):
+        """Write a line."""
+        self._line = line
+
+    def read(self):
+        """Read a line."""
+        return self._line
+
+
+class CSVSerializer(BaseSerializer):
+    """CSV serializer for records.
+
+    Note: This serializer is not suitable for serializing large number of
+    records.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize CSVSerializer.
+
+        :param csv_excluded_fields: list of paths of the fields that
+                                    should be excluded from the final output
+        :param csv_included_fields: list of paths of the only fields that
+                                    should be included in the final output
+        :param header_separator: separator that should be used when flattening
+                                 nested dictionary keys
+        :param collapse_lists: prevent lists being expanded into many columns
+                               and instead newline seperated fields
```

Comment:
minor: we should test these

Reply:
https://github.com/inveniosoftware/flask-resources/issues/125